go/src/koding/klient/machine/mount/notify/notify.go

```diff
@@ -0,0 +1,43 @@
+package notify
+
+import (
+	"context"
+
+	"koding/klient/machine/index"
+	"koding/klient/machine/mount"
+)
+
+// BuildOpts represents the context that can be used by external notifiers to
+// build their own type. Built notifier should only read from provided indexes
+// and, if changes occur, commit observed changes using Cache interface.
+type BuildOpts struct {
+	MountID mount.ID    // identifier of synced mount.
+	Mount   mount.Mount // single mount with absolute paths.
+
+	Cache    Cache  // index file system change consumer.
+	CacheDir string // absolute path to locally cached files.
+
+	RemoteIdx *index.Index // known state of remote index.
+	LocalIdx  *index.Index // known state of local index.
+}
+
+// Builder represents a factory method which external notifiers must implement
+// in order to create their instances.
+type Builder interface {
+	// Build uses provided build options to create Notifier instance.
+	Build(opts *BuildOpts) (Notifier, error)
+}
+
+// Notifier is an interface which must be implemented by external notifiers.
+type Notifier interface {
+	// Close cleans up notifier resources, if any.
+	Close()
```

Comment:
Would it be possible to have `Close() error` here? If notifier, for whatever reason, needs to implement io.Closer interface, it would conflict with this one. Or rename it to `Stop()`.